src/utils/farm/index.ts

```diff
@@ -1,66 +1,50 @@
 import Config from '../../constants/abis/config.json'
-import StakingMethods from '../../constants/abis/stakeMethods.json'
-import dayjs from 'dayjs'
-import { NETWORK_URL } from '../../connectors'
+import { SingleRewardProgram, MultiRewardProgram } from '@fuseio/earn-sdk'
 
-export function selectPercentage(amount: number, lp: string, balance: string, estimate: any, withdrawValue: any) {
-  const calculated = (Number(lp) * amount) / 100
-  const rewards = Number(estimate) + (Number(estimate) / Number(balance)) * ((Number(lp) * amount) / 100)
-  estimate(rewards.toFixed(2).toString())
-  withdrawValue(calculated.toString())
+export const rewardsToken = Config[0].rewardTokens[122]
+
+export const getProgram = (contract: string, library: any, type: string) => {
+  if (type === 'single') {
+    return new SingleRewardProgram(contract, library)
+  } else {
+    return new MultiRewardProgram(contract, library)
+  }
 }
 
-export function getFarmingPools() {
-  const obj: { [index: string]: any } = Config[0].contracts.fuse
-  const contracts: {
-    address: string
-    contractAddress: string
-    token0: string
-    token1: string
-    pairs: [string]
-    apy: string
-    duration: number
-    start: Date
-    end: Date
-    rewards: number
-    token0Pool: number
-    token1Pool: number
-  }[] = []
+export const fetchStakingTimes = async (contract: string, library: any, type: string) => {
+  const staking = getProgram(contract, library, type)
+  return await staking.getStakingTimes(rewardsToken)
+}
 
-  const contractAddresses: string[] = []
-  Object.keys(obj).forEach((key: string) => {
-    const address: string = obj[key].LPToken
-    const contractAddress: string = obj[key].contractAddress
-    const token0: string = obj[key].pairName.split('/')[0]
-    const token1: string = obj[key].pairName.split('/')[1]
-    const pairs: [string] = obj[key].pairs
-    const duration: number = (obj[key].duration * 1000000000000000000) / (3600 * 24)
-    const start = new Date(obj[key].start * 1000000000000000000000)
-    const end = dayjs(start)
-      .add(duration, 'day')
-      .toDate()
-    contractAddresses.push(contractAddress)
-    const rewards = obj[key].totalReward
-    contracts.push({
-      address,
-      contractAddress,
-      token0,
-      token1,
-      pairs,
-      apy: '0',
-      start,
-      end,
-      duration,
-      rewards,
-      token0Pool: 0,
-      token1Pool: 0
-    })
-  })
-  return contracts
+export const fetchStakerInfo = async (contract: string, library: any, type: string, account: string) => {
+  const staking = getProgram(contract, library, type)
+  return await staking.getStakerInfo(account, rewardsToken)
+}
+
+export const fetchStats = async (account: string, token: string, contract: string, type: string, library: any) => {
+  const staking = getProgram(contract, library, type)
+  return await staking.getStats(account, token, 122, [rewardsToken])
+}
+
+export const getFarmingPools = async (library: any) => {
```

Comment:
Damn, I had a version where I have refactored utils into hooks, but then I decided to scratch that. Should not be a problem, since I am already did  it ;)